lib/utils.ts

```
@@ -4,3 +4,40 @@ import { twMerge } from "tailwind-merge"
   return twMerge(clsx(inputs))
 }
+
+// Validate and normalize external URLs to http(s) only.
+// Returns a normalized href string or null if invalid/unsafe.
+export function toSafeExternalHref(input?: unknown): string | null {
+  try {
+    if (typeof input !== 'string') return null;
+    const raw = input.trim();
+    if (!raw) return null;
+    // Reject any ASCII control characters (U+0000–U+001F, U+007F)
+    if (/[\u0000-\u001F\u007F]/.test(raw)) return null;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)